sqlmesh/dbt/builtin.py

```diff
@@ -306,9 +306,15 @@ def create_builtin_globals(
         {k: builtin_globals.get(k) for k in ("ref", "source", "config")}
     )
 
+    target = jinja_globals.get("target")
+    dialect = target.type if target else None
+
     if engine_adapter is not None:
         adapter = RuntimeAdapter(
-            engine_adapter, jinja_macros, jinja_globals={**builtin_globals, **jinja_globals}
+            engine_adapter,
+            jinja_macros,
+            jinja_globals={**builtin_globals, **jinja_globals},
+            dialect=dialect,
```

Comment:
I did it for consistency between the ParseTime and RunTime rendering. Probably 6 of one, half dozen of another. Do you have a preference?

Reply:
I think it's important to use the engine adapter's dialect in the runtime adapter because the engine used for evaluation may potentially differ from the initial target.